recipes-extended/xen/files/libxl-RFC-5of7-Handle-Linux-stubdomain-specific-QEMUoptions.patch

```diff
@@ -64,23 +64,40 @@ PATCHES
                                    NULL);
                  ioemu_nics++;
              }
-@@ -912,6 +915,7 @@ static int libxl__build_device_model_arg
+@@ -1147,6 +1150,7 @@ static int libxl__build_device_model_arg
+     int i, connection, devid;
+     uint64_t ram_size;
      const char *path, *chardev;
-     char *user = NULL;
-     struct passwd *user_base, user_pwbuf;
 +    bool is_stubdom = libxl_defbool_val(b_info->device_model_stubdomain);
  
      dm_args = flexarray_make(gc, 16, 1);
      dm_envs = flexarray_make(gc, 16, 1);
-@@ -922,24 +926,18 @@ static int libxl__build_device_model_arg
+@@ -1157,38 +1161,18 @@ static int libxl__build_device_model_arg
                        "-xen-domid",
                        GCSPRINTF("%d", guest_domid), NULL);
  
 -    flexarray_append(dm_args, "-chardev");
--    flexarray_append(dm_args,
--                     GCSPRINTF("socket,id=libxl-cmd,"
--                                    "path=%s/qmp-libxl-%d,server,nowait",
--                                    libxl__run_dir_path(), guest_domid));
+-    if (state->dm_monitor_fd >= 0) {
```

Comment:
Why were these changes needed?  Can we use any of the work done by Qubes with upstream Xen 4.12 for Linux stubdoms?

Reply:
It's hard to tell from this patch diff, but this was just a contextual adjustment. The end result is the same.